### ISSUE TITLE
Question: how to WLS2 + React Native + Flipper + Windows 10

### ISSUE THREAD

**Issue by angelod1as** [facebook/flipper#1291]
Hey!

I'm running WSL2 (Ubuntu 20.04 LTS) in Windows 10, developing React + React Native applications. [After following this tutorial](https://gist.github.com/angelod1as/3d3755a2a6714398203442b760006859), I managed to make Android Studios + WSL2 work perfectly... but Flipper doesn't.

Flipper shows me React DevTools and Logs. It doesn't show anything on Hermes Debugger (it keeps saying to reconnect DevTools because Websocket disconnected, even with the other tabs working).

That said, Flipper doesn't show the app tab nor the network one (this one is a must in my project).

What might I be doing wrong?

**Comment by angelod1as** [comment]
I believe I'm already doing this though these lines in the tutorial:

```
iex "netsh interface portproxy delete v4tov4 listenport=8081 listenaddress=127.0.0.1" | out-null;
$WSL_CLIENT = bash.exe -c "ip addr show eth0 | grep -oP '(?<=inet\s)\d+(\.\d+){3}'";
$WSL_CLIENT -match '\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}';
$WSL_CLIENT = $matches[0];
iex "netsh interface portproxy add v4tov4 listenport=8081 listenaddress=127.0.0.1 connectport=8081 connectaddress=$WSL_CLIENT"
```

If this isn't it, could you please help me do it?

**Comment by passy** [comment]
@angelod1as Sorry, I might be just missing something but where does this forward port 8001?

**Comment by angelod1as** [comment]
So, I think I believed it wrong. How can I do this port forwarding? If you be so kind of giving me instructions, I'll follow them and post my answers here right away.

**Comment by passy** [comment]
I'm sorry, I have absolutely no idea. As I've said, I have not used Windows in anger for probably a decade by now.

**Comment by angelod1as** [comment]
I'm no longer using WLS2 so this solution is no longer necessary.

**Comment by luizwhite** [comment]
I really want to know how to do this, I have a working React Native / WSL2 envirnoment, and a "kind of working" flipper.
The thing is that Flipper executes adb but i am already running an adb server, so i dunno the correct way to handle that.
I executed before opening Flipper `adb reverse tcp:8096 tcp:8096 && adb reverse tcp:8097 tcp:8097`

And i can see the device, the app, and the network jsut fine
But I don't know, it seems buggy, maybe i need to do something else

Is there a plan to Flipper create a tutorial for WSL2 users?

**Comment by alloystory** [comment]
just figured out the solution to get Hermes Debugger running on WSL
TLDR; run metro bundler with `yarn start --host 0.0.0.0` instead of `yarn start`

Flipper [relies on the response](https://github.com/facebook/flipper/blob/0b2e5f645467bd0ee886d651ebf9cc7e5752ffbc/desktop/plugins/hermesdebuggerrn/index.tsx#L90) of `localhost:8081/json` to find the `devtoolsFrontendUrl`. This url is used to show the Hermes Debugger.

Example response of `localhost:8081/json`
```json
[
  {
    "id": "1-1",
    "description": "myapp.local",
    "title": "Hermes React Native",
    "faviconUrl": "https://reactjs.org/favicon.ico",
    "devtoolsFrontendUrl": "chrome-devtools://devtools/bundled/inspector.html?experiments=true&v8only=true&ws=%5B%3A%3A%5D%3A8081%2Finspector%2Fdebug%3Fdevice%3D1%26page%3D1",
    "type": "node",
    "webSocketDebuggerUrl": "ws://[::]:8081/inspector/debug?device=1&page=1",
    "vm": "Hermes"
  }
]
```

However, by default, metro will show an IPv6 address in `devtoolsFrontendurl` (note the `[::]`), which [cannot be accessed](https://github.com/microsoft/WSL/issues/4518) outside of WSL.

After some digging through the metro repo, it looks like by starting the metro bundler with an IPv4 host address, we force metro to show an IPv4 address in `devtoolsFrontendUrl`. This should be accessible outside of WSL, allowing Hermes Debugger to work in Flipper.

**Comment by Manish2612** [comment]
@alloy Thanks! Your solution worked for me.
